docs/UAA-Client-Authentication.md

```diff
@@ -0,0 +1,83 @@
+of public keys and this set can contain many keys because each key has its own kid (key id). The keys can be stored in UAA own persistency or with
+a dynamic token key URI. OIDC has defined the parameter jwks_uri for this already. The structure of the keys is defined with [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
+UAA provides its own jwks_uri with endpoint /token_keys. The content of this endpoint is [JWKS](https://datatracker.ietf.org/doc/html/rfc7517#section-5).
+
+The new parameters for UAA clients are:
+
+* jwks_uri
+* jwks
+
+This should allow a continues trust between a UAA to UAA communication, e.g. using own UAA instances or within a UAA using different zones.
+
+### tls_client_auth (Planned Feature)
+Not yet defined a release date.
+
+## Configs
+Here is a brief example of the `clients` section:
+```yaml
+oauth:
+  clients:
+    uaa-trust-uri:
+      authorities: scim.zones,uaa.zones.read,uaa.zones.write,uaa.admin,clients.read,clients.write,clients.secret,zones.read,zones.uaa.admin
+      authorized-grant-types: client_credentials
+      id: uaa_trust
+      scope: none
+      jwks_uri: http://localhost:8080/uaa/token_keys
+    uaa-trust-keys:
+      authorities: scim.zones,uaa.zones.read,uaa.zones.write,uaa.admin,clients.read,clients.write,clients.secret,zones.read,zones.uaa.admin
+      authorized-grant-types: client_credentials
+      id: uaa_trust
+      scope: none
+      jwks: |
+        {
+          "keys": [
+            {
+              "kty": "RSA",
+              "e": "AQAB",
+              "use": "sig",
+              "kid": "legacy-token-key",
+              "alg": "RS256",
+              "n": "qMClJXznycV2bQ1pFbN8W-AWSYhpS2MVAGhkWNlmxv2Ix0_-n6zjivjdoxcq7RJR4kVycoVeD07DiWElYSnQLdeQPgKAcBiwilR30UyyDTKcqDQQ5rkCg2ONlwV0aMsg74KaXeXsV653ASs3FYEtuS1aD_Db5-FyXF8HkHo8xy19NUnqsDWQnh1Hhklynxu2tvW0fw2oDE1pwNl-WLEVPtlcpCtf4VSv-GawtBiI6xmYsGBMC9w29ESHFqPw0NSCRhlyJf6rDBNH_766mzK_vEzA4rzGTBEUqDxTg_8JpRhh9D3qljSsmqCtpQoloOAaUKCqSJb_hKPspe-7r9cYmw"
+            }
+          ]
+        }
+```
+The example config above with jwks_uri enables continues trust to a running uaa.
+
+### Developer implementation
+As deverloper you should use the [UAA documention](https://docs.cloudfoundry.org/api/uaa/version/76.23.0/index.html#token). There is a description
+about the new parameters client_assertion and client_assertion_type. In addition, you can check in the retrieved access_token tokens for the existence 
+of claim client_auth_method with value private_key_jwt, (client_auth_method=private_key). This claim should guarantee the used method of client 
+authentication. Tokens without this claim are authenticated with secrets. There might be use-cases where a stronger authentication mechanism is 
+required.
+
+### Production use
+
+The introduction of private_key_jwt for production is planned with end of Q4/2023. The use for other purposes is given with a release of UAA starting.
```

Comment:
> The introduction of private_key_jwt for production

I do not know what this means. Are we talking about some downstream component/product that contains or depends on UAA"?

Reply:
> At the beginn we (SAP) will run it

Right, I think that I did guess that. When we say "for production", it gives me impression that there is the one and only production software for UAA. I think that it will be less confusing if we say "for a production" instead.

Reply:
done now ?